lib/Sema/CSDiag.cpp

```diff
@@ -1783,15 +1709,6 @@ void FailureDiagnosis::diagnoseAmbiguity(Expr *E) {
     return;
   }
 
-  // A DiscardAssignmentExpr (spelled "_") needs contextual type information to
-  // infer its type. If we see one at top level, diagnose that it must be part
-  // of an assignment so we don't get a generic "expression is ambiguous" error.
-  if (isa<DiscardAssignmentExpr>(E)) {
-    diagnose(E->getLoc(), diag::discard_expr_outside_of_assignment)
-      .highlight(E->getSourceRange());
-    return;
-  }
-  
   // Diagnose ".foo" expressions that lack context specifically.
   if (auto UME =
         dyn_cast<UnresolvedMemberExpr>(E->getSemanticsProvidingExpr())) {
```

Comment:
Btw, I think this should be obsolete as well because diagnostics for `_ = .foo` have been ported already as missing base type for unresolved member.

Reply:
It isn't yet :/ it's still handling cases like this, where the solver does find multiple solutions with fixes but it isn't yet diagnosed in `diagnoseAmbiguityWithFixes` :

```swift
func f(_: Int) {}
func f(_: String) {}

f(.nothing)
```

Reply:
Ah, that comment on top threw me off :)